cmd/main.go

```diff
@@ -62,6 +61,7 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
+//nolint:funlen // Avoid splitting the main function in multiple functions to avoid changing the retcode logic for metrics shutdown
```

Comment:
I am a bit afraid of the `funlen` linter in the long term. 
We often are forced to disable it or create many functions to make it happy, where it would be simpler to have a single function for readability.
Should we keep it around?

Reply:
let's see how often it gets in our way. We can then disable on a per-function basis. Eventually we can completely turn it off

Reply:
I'm ok with punting on it.